javalib/src/main/scala/java/util/Formatter.scala

```diff
@@ -598,13 +598,13 @@ final class Formatter private (private[this] var dest: Appendable,
   }
 
   private def formatNumericString(localeInfo: LocaleInfo, flags: Flags,
-      width: Int, str: String): Unit = {
+      width: Int, str: String, maybeBasePrefix: Option[String] = None): Unit = {
```

Comment:
If we keep using `formatNumericString`, this parameter could be of type `String` with a default value of `""`. This would avoid the boxing in the Option.

(Also we try to avoid Scala stuff in the javalib.)